test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,347 @@
+import { expect } from 'chai';
+import { Writable } from 'stream';
+
+import {
+  MongoLogger,
+  MongoLoggerMongoClientOptions,
+  MongoLoggerOptions,
+  SeverityLevel
+} from '../../src/mongo_logger';
+
+class BufferingStream extends Writable {
+  buffer: string[] = [];
+
+  constructor(options = {}) {
+    super({ ...options, objectMode: true });
+  }
+
+  override _write(chunk, encoding, callback) {
+    this.buffer.push(chunk);
+    callback();
+  }
+}
+
+describe('meta tests for BufferingStream', function () {
+  it('the buffer is empty on construction', function () {
+    const stream = new BufferingStream();
+    expect(stream.buffer).to.have.lengthOf(0);
+  });
+  it('pushes messages to the buffer when written to', function () {
+    const stream = new BufferingStream();
+    stream.write('message');
+    expect(stream.buffer).to.deep.equal(['message']);
+  });
+});
+
+describe('class MongoLogger', function () {
+  describe('#constructor()', function () {
+    it('assigns each property from the options object onto the logging class', function () {
+      const componentSeverities: MongoLoggerOptions['componentSeverities'] = {
+        command: 'alert'
+      } as any;
+      const stream = new Writable();
+      const logger = new MongoLogger({
+        componentSeverities,
+        maxDocumentLength: 10,
+        logDestination: stream
+      });
+
+      expect(logger).to.have.property('componentSeverities', componentSeverities);
+      expect(logger).to.have.property('maxDocumentLength', 10);
+      expect(logger).to.have.property('logDestination', stream);
+    });
+  });
+
+  describe('static #resolveOptions()', function () {
+    describe('componentSeverities', function () {
+      const components = [
+        'MONGODB_LOG_COMMAND',
+        'MONGODB_LOG_TOPOLOGY',
+        'MONGODB_LOG_SERVER_SELECTION',
+        'MONGODB_LOG_CONNECTION'
+      ];
+      const mapToInternalRepresentation = (component: string) => {
+        const options: Record<string, string> = {
+          MONGODB_LOG_COMMAND: 'command',
+          MONGODB_LOG_TOPOLOGY: 'topology',
+          MONGODB_LOG_SERVER_SELECTION: 'serverSelection',
+          MONGODB_LOG_CONNECTION: 'connection'
+        };
+        return options[component];
+      };
+
+      context('MONGODB_LOG_ALL', () => {
+        context('when a default is provided', () => {
+          it('sets default to the provided value', () => {
+            const options = MongoLogger.resolveOptions(
+              { MONGODB_LOG_ALL: SeverityLevel.ALERT },
+              {}
+            );
+            expect(options.componentSeverities).to.have.property('default', SeverityLevel.ALERT);
+          });
+        });
+        context('when no value is provided', () => {
+          it('sets default to off', () => {
+            const options = MongoLogger.resolveOptions({ MONGODB_LOG_ALL: SeverityLevel.OFF }, {});
```

Comment:
this is actually not testing when no value is provided, it's testing when the explicit `OFF` value is provided